pkgs/applications/networking/instant-messengers/session-desktop/default.nix

```diff
@@ -18,39 +15,32 @@ let
   appimage = appimageTools.wrapType2 {
     inherit version pname src;
   };
-  appimage-contents = appimageTools.extractType2 {
+  appimageContents = appimageTools.extractType2 {
     inherit version pname src;
   };
 in
 stdenvNoCC.mkDerivation {
   inherit version pname;
   src = appimage;
 
-  nativeBuildInputs = [ copyDesktopItems makeWrapper ];
-
-  desktopItems = [
-    (makeDesktopItem {
-      name = "Session";
-      desktopName = "Session";
-      comment = "Onion routing based messenger";
-      exec = "${appimage}/bin/session-desktop-${version}";
-      icon = "${appimage-contents}/session-desktop.png";
-      terminal = false;
-      type = "Application";
-      categories = [ "Network" ];
-    })
-  ];
+  nativeBuildInputs = [ makeWrapper ];
 
   installPhase = ''
     runHook preInstall
 
-    mv bin/session-desktop-${version} bin/session-desktop
+    mv bin/${pname}-${version} bin/${pname}
 
     mkdir -p $out/
     cp -r bin $out/bin
 
-    wrapProgram $out/bin/session-desktop \
-      --add-flags "\''${NIXOS_OZONE_WL:+\''${WAYLAND_DISPLAY:+--ozone-platform-hint=auto --enable-features=WaylandWindowDecorations}}"
+    mkdir -p $out/share/${pname}
+    cp -a ${appimageContents}/usr/share/icons $out/share/
+    install -Dm 644 ${appimageContents}/${pname}.desktop -t $out/share/applications/
+
+    substituteInPlace $out/share/applications/${pname}.desktop --replace "AppRun" "${pname}"
+
+    wrapProgram $out/bin/${pname} \
+      --add-flags "\''${NIXOS_OZONE_WL:+\''${WAYLAND_DISPLAY:+--ozone-platform=wayland --enable-features=WaylandWindowDecorations,UseOzonePlatform}}"
```

Comment:
This and the old one ran fine for me on wayland, so lgtm

Reply:
It seems like `ozone-platform-hint` was only added to Electron versions 18+ (and not on version 17.4.10 which is what Session is using (shame))

https://github.com/electron/electron/pull/34937

I could not get native Wayland to work with the hint flag, but it works with the explicit flag. I wonder what your environment is doing that causes this to happen.